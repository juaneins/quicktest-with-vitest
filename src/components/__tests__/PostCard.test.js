import axios from 'axios';
import { mount, flushPromises } from '@vue/test-utils';
import { describe, test, vi, expect } from 'vitest';
import PostCard from '../PostCard.vue';

const mockPost = {
  userId: 1,
  id: 1,
  title: 'Post title 1',
  body: 'Simple post body...'
};

describe('Post card Component', () => {
  test('can fetch and display a post', async () => {
    vi.spyOn(axios, 'get').mockResolvedValueOnce({ data: mockPost });

    const wrapper = mount(PostCard);
    expect(wrapper.html()).toContain('Loading...');

    await flushPromises();

    expect(wrapper.find('[data-testid="post-title"]').text()).toBe(mockPost.title);

    expect(wrapper.find('[data-testid="post-body"]').text()).toBe(mockPost.body);
  });

  test('can display an error message if fetching a post fails', async () => {
    vi.spyOn(axios, 'get').mockRejectedValueOnce(new Error('Error Ocurred'));

    const wrapper = mount(PostCard);
    expect(wrapper.html()).toContain('Loading...');

    await flushPromises();

    expect(wrapper.find('[data-testid="error-message"]').text()).toBe('Error Ocurred');
  });
});
