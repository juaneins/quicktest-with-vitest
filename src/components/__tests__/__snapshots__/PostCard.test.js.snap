// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Post card Component > created post render correctly 1`] = `
"<div title="Test Post" body="test body post...">
  <!--v-if-->
  <p data-testid="loader">Loading...</p>
  <!--v-if-->
</div>"
`;
